File: tests/go/switch_type_tagless/switch_fallthrough.go
// vybe-test: go/switch_type_tagless/switch_fallthrough
// origin: languages/go/tests/go/test_switch_type_tagless.rs

package main
import "fmt"
func __check(got string, want string) {
	if got != want {
		fmt.Println("FAIL: want [" + want + "] got [" + got + "]")
		panic("assertion failed")
	}
}

func main() { x := 1
switch x { case 1: __check(fmt.Sprint("a"), "a")
fallthrough
case 2: __check(fmt.Sprint("b"), "b") } }

File: tests/js/missing_features/switch_fallthrough.js
// vybe-test: js/missing_features/switch_fallthrough
// origin: languages/js/tests/js/js_missing_features_test.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

let x = 1;
        let result = "";
        switch (x) {
            case 1:
                result += "one";
            case 2:
                result += "two";
            case 3:
                result += "three";
        }
        __check(__line(result), "onetwothree");
